streaming/base/shared.py

```diff
@@ -268,3 +272,79 @@ def cleanup(self):
             pass
         finally:
             resource_tracker.unregister = original_rtracker_unreg
+
+
+def get_shm_prefix(my_locals: List[str], world: World) -> Tuple[str, SharedMemory]:
+    """Register or lookup our shared memory prefix.
+
+    Args:
+        my_locals (List[str]): Local working dir of each stream, relative to /. We need to verify
```

Comment:
What happens when `my_locals` is `[]` ? Meaning if someone is not using `Streams()`!

Reply:
we have a check in streaming dataset init for that (also using just remote/local is rolled up into a stream):

```
        if bool(streams) == (bool(remote) or bool(local)):
            raise ValueError(
                'You must provide either "streams" or "remote"/"local", but not both -- ' +
                'that would be confusing')
```